File: java/src/Ice/src/main/java/Ice/BatchRequestInterceptor.java
// **********************************************************************
//
// Copyright (c) 2003-2018 ZeroC, Inc. All rights reserved.
//
// This copy of Ice is licensed to you under the terms described in the
// ICE_LICENSE file included in this distribution.
//
// **********************************************************************

package Ice;

/**
 * Base interface for listening to batch request queues.
 **/
public interface BatchRequestInterceptor
{
    /**
     * Called by the Ice runtime when a batch request is about to be
     * added to the batch request queue of a proxy or connection.
     *
     * The implementation of this method must call enqueue() on the
     * request to confirm its addition to the queue, if not called
     * the request isn't added to the queue. The implementation can
     * raise an Ice local exception to notify the caller of a failure.
     *
     * @param request The request to be added.
     * @param queueBatchRequestCount The number of requests in the batch.
     * @param queueBatchRequestSize The number of bytes in the batch.
     **/
    void enqueue(Ice.BatchRequest request, int queueBatchRequestCount, int queueBatchRequestSize);
}

File: python/python/Ice/BatchRequestInterceptor.py
# Copyright (c) ZeroC, Inc.

from abc import ABC, abstractmethod

class BatchRequestInterceptor(ABC):
    """
    Base class for batch request interceptors.

    A subclass must define the `enqueue` method.
    """

    @abstractmethod
    def enqueue(self, request, queueCount, queueSize):
        """
        Invoked when a request is batched.

        Parameters
        ----------
        request : IcePy.BatchRequest
        queueCount : int
            The number of requests currently in the queue.
        queueSize : int
            The number of bytes consumed by the requests currently in the queue.
        """
        pass
